int32 allied_type
bool enabled
---
int32 error